code/src/components/GameBoard.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+import "nes.css/css/nes.min.css"
+
+import game, { continueGame } from 'reducers/game'
+
+import LoadingIndicator from './LoadingIndicator'
+import EndOfGame from './EndOfGame'
+import { GameBoardContainer, Header, ActionContainer, GoBackButton, Action, ActionDescription } from './styled-components/GameBoardStyling'
+
+const GameBoard = () => {
+    const isLoading = useSelector((store) => store.ui.isLoading)
+    const actions = useSelector((store) => store.game.gameStatus.actions)
+    const currentDescription = useSelector((store) => store.game.gameStatus)
+    const userName = useSelector((store) => store.game.username)
+    const history = useSelector((store) => store.game.history)
+
+    const dispatch = useDispatch()
+    
+    if (actions.length === 0) {
+        return (
+            <EndOfGame />
+        )
+    }
+    
+    return (
+        <>
+            {isLoading 
+                ? <LoadingIndicator />
+                : <GameBoardContainer>
+                    <div class="nes-container is-rounded is-dark">
+                        <Header>{currentDescription.description}</Header>
+                        {actions?.map((action, index) => {
+                            return(
+                            <Action>
+                                <div key={index} class="nes-container is-rounded is-dark">
+                                    <ActionContainer>
+                                        <ActionDescription>{action.description}</ActionDescription>
+                                        <button 
+                                            type="button" 
+                                            class="nes-btn is-success"
+                                            aria-label='direction button'
+                                            onClick={() => dispatch(continueGame(action.direction, userName))}
+                                        >
+                                            {action.direction}
+                                        </button>
```

Comment:
Interesting that you chose to pass the username in the dispatch, we did it in a different way where we got it directly from state. But that was kind of a hassle so this way maybe is a bit smoother 👍